com.unity.sg2/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -322,6 +334,39 @@ public override INodeModel DuplicateNode(INodeModel sourceNode, Vector2 delta, I
             return pastedNodeModel;
         }
 
+        public override TDeclType DuplicateGraphVariableDeclaration<TDeclType>(TDeclType sourceModel, bool keepGuid = false)
+        {
+            var uniqueName = sourceModel.Title;
+            var copiedVariable = sourceModel.Clone();
+            copiedVariable.GraphModel = this;
+            if (keepGuid)
+                copiedVariable.Guid = sourceModel.Guid;
+            copiedVariable.Title = GenerateGraphVariableDeclarationUniqueName(uniqueName);
+
+            // Need to get a reference to the original variable declaration model
+            // to get stored default value of the constant
+            var originalVariable = VariableDeclarations.FirstOrDefault(model => model.Guid == sourceModel.Guid);
+            if (originalVariable != null)
+            {
+                // Initialize CLDS data prior to initializing the constant
+                ShaderGraphStencil.InitVariableDeclarationModel(copiedVariable, copiedVariable.InitializationModel);
+                copiedVariable.CreateInitializationValue();
+                copiedVariable.InitializationModel.ObjectValue = originalVariable.InitializationModel.ObjectValue;
+            }
+
+            AddVariableDeclaration(copiedVariable);
+
+            if (sourceModel.ParentGroup != null && sourceModel.ParentGroup.GraphModel == this)
+                sourceModel.ParentGroup.InsertItem(copiedVariable, -1);
```

Comment:
Should pasted variables go to the top of the list? Kinda seems like they should go to the bottom. Maybe just dbl check w/design if possible.

Speaking of which-- does Cut/Paste work? It's funny that when you _cut_ you don't get the same issues with deletion. Are the cut variables removed from GTF model during the cut but still in GraphHandler?

This is a weird case that will end up needing some strange tests to really be certain on.

Reply:
Huh you're right! I think we should raise this one up with GTF, will bring it up at our next sync